com.ibm.team.build.hjplugin/src/main/java/com/ibm/team/build/internal/hjplugin/RTCChangeLogParser.java

```diff
@@ -32,87 +29,86 @@
 
 public class RTCChangeLogParser extends ChangeLogParser {
     private static final Logger LOGGER = Logger.getLogger(RTCChangeLogParser.class.getName());
-
+	
 	@Override
-	public ChangeLogSet<? extends Entry> parse(AbstractBuild build,
-			File changelogFile) throws IOException, SAXException {
-		FileInputStream inputStream = new FileInputStream(changelogFile);
-        CharsetDecoder decoder = Charset.forName("UTF-8").newDecoder().onMalformedInput(CodingErrorAction.REPORT).onUnmappableCharacter(CodingErrorAction.REPORT); //$NON-NLS-1$
-		Reader reader = new InputStreamReader(inputStream, decoder);
-		return parse(build, reader);
+	public ChangeLogSet<? extends Entry> parse(Run build,
+			RepositoryBrowser<?> browser, File changeLogFile)
+			throws IOException, SAXException {
+		FileReader changeLogReader = new FileReader(changeLogFile);
+		try {
+			return parse(build, browser, changeLogReader);
+		} finally {
+			changeLogReader.close();
+		}
 	}
 	
-	public ChangeLogSet<? extends Entry> parse(AbstractBuild build,
-			Reader changelogReader) throws IOException, SAXException {
+	public ChangeLogSet<? extends Entry> parse(Run build,
+			RepositoryBrowser<?> browser, Reader changeLogReader)
+			throws IOException, SAXException {
 
-		try {
-			RTCChangeLogSet result = new RTCChangeLogSet(build);
-			Digester2 digester = getDigester();
-			digester.push(result);
-	
-			digester.addSetProperties("changelog"); //$NON-NLS-1$
```

Comment:
BTW I recommend trying to minimize diff lines, even at the cost of leaving indentation less than perfect (can always reindent in a separate PR). This is really hard to follow (need to append `?w=1` to URL).